pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1105,6 +1105,36 @@ cdef class _Timedelta(timedelta):
                .format(td=components, seconds=seconds))
         return tpl
 
+    # ----------------------------------------------------------------
+
+    def _round(self, freq, rounder):
+        cdef:
+            int64_t result, unit
+
+        unit = to_offset(freq).nanos
+        result = unit * rounder(self.value / float(unit))
+        return Timedelta(result, unit='ns')
+
+    def floor(self, freq):
+        """
+        return a new Timedelta floored to this resolution
+
+        Parameters
+        ----------
+        freq : a freq string indicating the flooring resolution
+        """
+        return self._round(freq, np.floor)
+
+    def ceil(self, freq):
```

Comment:
why are you not *also* moving ``round``? either they should *all* be in the base class or none, otherwise the code is too confusing. Though having the user interface in the python class is prob ok as well (IOW the _ leading functions in the base class is ok, while the others remain). I suspect this is just plain inconsistent across all of the scalars. Rather than just wholesale changes, can you see what makes sense first.

Reply:
I did on the first pass and got errors complaining that `round` doesn't take any keyword arguments (which makes no sense).  No idea what was going on there.  Regardless, this part is getting reverted.